### PR TITLE
Claymores & Mine Tweaks

### PR DESCRIPTION
**Pull request by Sun-Soaked**
## About The Pull Request

TODO:

- [x] Actually implement claymores (should we let players have them?)(Cargo?) 

- [x] Remove weird sprite shim I made when world icons are merged, add world icons to mines

- [x] Sprites (this time we actually need them since the placeholder ones are tgmc & I didn't ask for permission)

Adds Mines that explode when an object passes in front of them & fire an ammo casing in that direction.
Adds claymores as an implemented example of this- short range, shreds people at point blank.
Adds World Icons to normal Mines

I had to do some tweaking behind the scenes to implement mobless casing firing- but I'm pleased to say that fire_casing() can now deal emotionally with being handed a null user.

Also minorly tweaks explosion devastation, so heavy explosions can only strip tiles that have already been damaged
You can still immediately strip tiles with devastating explosions

## Why It's Good For The Game
FRONT TOWARDS ENEMY

## Changelog

:cl:
add: Directional Mines, implemented as Claymores.
add: Mines now have World Icons
balance: explosions now only strip to space when a tile is already damaged
fix: Prox mines explode from slightly further away & no longer activate through walls.
fix: Multiple mines can no longer be stacked on a tile
code: fire_casing() can now be called by non-mobs, allowing objects to shoot casings.
/:cl:

### CODE REVIEW

**Review by FalloutFalcon — APPROVED** [review]
code seems fine!

**Comment by Sun-Soaked** [comment]
this could use a testmerge now that they're accessible via cargo, I want to see how the balance shakes out

**Comment by Sun-Soaked** [comment]
urge to kill mojave world icons grows

**Comment by thgvr** [comment]
Not really a fan of how small the landmines end up, they need to be large. Claymores maybe passable

**Comment by Sun-Soaked** [comment]
> Not really a fan of how small the landmines end up, they need to be large. Claymores maybe passable

I budgeted in the world icons so we need to use them somewhere for the tax exemption

**Comment by Sun-Soaked** [comment]
> Not really a fan of how small the landmines end up, they need to be large. Claymores maybe passable

actual problem is that claymores are still using placeholder tgmc sprites & I couldn't find the person who originally made them

**Comment by Anticept** [comment]
> * Sprites (this time we actually need them since the placeholder ones are tgmc & I didn't ask for permission)

Sprites are CC-BY-NC, no permission required.

**Comment by Sun-Soaked** [comment]
this is good to merge now btw

File: code/game/objects/items/devices/mines.dm [hunk 1]
```diff
@@ -526,6 +615,76 @@
...
 
+
+
+///Claymores
+///shrapnel based dir explosive, extreme short range
+///FRONT TOWARDS ENEMY
+/obj/item/mine/directional/claymore
+	name = "C-10 Claymore"
+	desc = "A compact anti-personnel device with a directional trigger that responds to movement. A faded sticker on the back reads \"FRONT TOWARDS ENEMY\"."
+	icon = 'icons/obj/world/landmine.dmi'
```
Comment (thgvr) [inline code/game/objects/items/devices/mines.dm:621]: do not do world icons for either of these, they should be visible.

**Comment by thgvr** [comment]
think you could add a red glow to armed mines so you can see them in the dark?

File: code/game/objects/items/devices/mines.dm (hunk 1 above)
Comment (Sun-Soaked) [reply code/game/objects/items/devices/mines.dm:621]: they are- I stuck the "planted" state from the original claymore sprite source in as the world icon

Comment (Sun-Soaked) [reply code/game/objects/items/devices/mines.dm:621]: normal landmine world icons aren't implemented rn

Comment (thgvr) [reply code/game/objects/items/devices/mines.dm:621]: Then just remove the world icon stuff. It looks like it's used.

**Review by FalloutFalcon — CHANGES_REQUESTED** [review]
Please don't put jokes in auto-doc! I would rather just not making it auto-doc if its not a proper explanation.

I would skim the auto-doc guide if you don't remember it.

**Comment by Sun-Soaked** [comment]
> Please don't put jokes in auto-doc! I would rather just not making it auto-doc if its not a proper explanation.
> 
> I would skim the auto-doc guide if you don't remember it.

bumping to remind you I made changes

**Review by FalloutFalcon — APPROVED** [review]

**Pull request merged**